stag-library/src/test/java/com/vimeo/stag/KnownTypeAdaptersTest.java

```diff
@@ -582,6 +702,22 @@ public void testForMapTypeAdapter() throws Exception {
         Utils.assertMapsEqual(intDummyMap, readValue1);
     }
 
+    @Test
+    public void mapTypeAdapterHandlesNullsCorrectly() throws Exception {
+        final HashMap<String, String> dummyMap = null;
+
+        TypeAdapter<HashMap<String, String>> mapTypeAdapter = new KnownTypeAdapters.MapTypeAdapter<>(TypeAdapters.STRING, TypeAdapters.STRING,
+                                                                                                     new KnownTypeAdapters.HashMapInstantiator<String, String>());
+
+        StringWriter stringWriter = new StringWriter();
```

Comment:
This `StringWriter` setup + adapter execution test code is repeated a lot throughout these tests. You might want to consider making some private local util method. However, since the adapter is the thing under test if you want to keep all of this test-specific, that's OK too. Your call.

Reply:
I noticed this as well, but I am going to hold off doing the refactoring to abstract this until I convert this class to Kotlin, because I tried it out and it creates more code in java and makes the tests more difficult to understand. Here's what I got:
```java
private <T> T writeAndReadValue(T value, Function2<JsonWriter, T, Unit> write, Function1<JsonReader, T> read) {
    StringWriter stringWriter = new StringWriter();
    write.invoke(new JsonWriter(stringWriter), value);
    String jsonString = stringWriter.toString();

    return read.invoke(new JsonReader(new StringReader(jsonString)));
}

// usage
@Test
public void testForPrimitiveByteTypeAdapter() throws Exception {
    byte value = 1;

    byte readValue = writeAndReadValue(value, new Function2<JsonWriter, Byte, Unit>() {
        @Override
        public Unit invoke(JsonWriter jsonWriter, Byte aByte) {
            try {
                KnownTypeAdapters.PrimitiveByteTypeAdapter.write(jsonWriter, aByte);
            } catch (IOException e) {
                throw new RuntimeException(e);
            }
            return Unit.INSTANCE;
        }
    }, new Function1<JsonReader, Byte>() {
        @Override
        public Byte invoke(JsonReader jsonReader) {
            try {
                return KnownTypeAdapters.PrimitiveByteTypeAdapter.read(jsonReader, (byte) 0);
            } catch (IOException e) {
                throw new RuntimeException(e);
            }
        }
    });

    assertEquals(value, readValue);
}
```